modules/shared/src/main/scala/io/constellationnetwork/env/JarSignature.scala

```diff
@@ -0,0 +1,42 @@
+package io.constellationnetwork.env
+
+import java.security.MessageDigest
+
+import cats.effect.Sync
+import cats.implicits.{catsSyntaxFlatMapOps, toFunctorOps}
+
+import io.constellationnetwork.security.hash.Hash
+import io.constellationnetwork.security.hex.Hex
+
+import fs2.Chunk
+import fs2.io.file.{Files, Path}
+
+object JarSignature {
+
+  def jarHash[F[_]: Sync: Files]: F[Hash] = {
+    val jarPath = JarSignature.getClass.getProtectionDomain.getCodeSource.getLocation.toURI
+    digestOf(Files[F].readAll(Path(jarPath.getPath)))
+  }
+
+  def digestOf[F[_]: Sync: Files](bytes: fs2.Stream[F, Byte]): F[Hash] = {
+    val chunkSize = 524288
+
+    digestInstance >>= { digest =>
+      bytes
+        .chunkN(chunkSize)
+        .evalMap(updateDigest(digest, _))
```

Comment:
maybe `map2` or `mapN`?

Reply:
sounds interesting, where do you suggest it?